include/material/mohr_coulomb.tcc

```diff
@@ -315,63 +355,79 @@ Eigen::Matrix<double, 6, 1> mpm::MohrCoulomb<Tdim>::compute_stress(
+  epsilon =
+      (1. / sqrt(3.)) * (trial_stress(0) + trial_stress(1) + trial_stress(2));
+  yield_type_trial =
+      this->check_yield(yield_function_trial, epsilon, rho, theta);
+  this->compute_df_dp(yield_type_trial, j2, j3, rho, theta, trial_stress,
+                      dF_dSigma_trial, dP_dSigma_trial, epds, softening_trial,
+                      ptr);
   // Check the epds
-  if(epds_last < peak_EPDS_ && epds > peak_EPDS_)softening_trial = 0;
-  if(yield_type_trial == 1)
-    lambda_trial = yield_function_trial(0) / ((dF_dSigma_trial.transpose() * de_).dot(dP_dSigma_trial.transpose()) + softening_trial);
+  if (epds_last < peak_EPDS_ && epds > peak_EPDS_) softening_trial = 0;
+  if (yield_type_trial == 1)
+    lambda_trial =
+        yield_function_trial(0) /
+        ((dF_dSigma_trial.transpose() * de_).dot(dP_dSigma_trial.transpose()) +
+         softening_trial);
   else
-    lambda_trial = yield_function_trial(1) / ((dF_dSigma_trial.transpose() * de_).dot(dP_dSigma_trial.transpose()) + softening_trial);
+    lambda_trial =
+        yield_function_trial(1) /
+        ((dF_dSigma_trial.transpose() * de_).dot(dP_dSigma_trial.transpose()) +
+         softening_trial);
   // Compute the correction stress
   double p_multiplier = 0.;
   Vector6d dP_dSigma_Final = Vector6d::Zero();
   bool PDS_update = false;
   // check if it is a load process or an unload process
   double dyfun_t = yield_function_trial(0) - yield_function(0);
   double dyfun_s = yield_function_trial(1) - yield_function(1);
-  if(yield_type !=0){
-    if(yield_type == 1 && dyfun_t > 0){
+  if (yield_type != 0) {
+    if (yield_type == 1 && dyfun_t > 0) {
       p_multiplier = lambda;
       dP_dSigma_Final = dP_dSigma;
       PDS_update = true;
-    }
-    else if(yield_type == 2 && dyfun_s > 0){
+    } else if (yield_type == 2 && dyfun_s > 0) {
       p_multiplier = lambda;
       dP_dSigma_Final = dP_dSigma;
       PDS_update = true;
     }
-  }
-  else if (yield_type_trial !=0) {
-      p_multiplier = lambda_trial;
-      dP_dSigma_Final = dP_dSigma_trial;
-      PDS_update = true;
+  } else if (yield_type_trial != 0) {
+    p_multiplier = lambda_trial;
+    dP_dSigma_Final = dP_dSigma_trial;
+    PDS_update = true;
   }
   // update stress (plastic correction)
-  Vector6d stress_update = trial_stress -  (p_multiplier * this->de_ * dP_dSigma_Final);
+  Vector6d stress_update =
+      trial_stress - (p_multiplier * this->de_ * dP_dSigma_Final);
   // compute plastic deviatoric strain
   Vector6d dstress = stress_update - stress;
   Vector6d dpstrain = dstrain - (this->de_.inverse()) * dstress;
-  if(Tdim == 2)dpstrain(4) = dpstrain(5) = 0.;
-  //Record the epds
+  if (Tdim == 2) dpstrain(4) = dpstrain(5) = 0.;
+  // Record the epds
   epds_last = epds;
-  //Update PDS if it is a load process
-  if(PDS_update){
+  // Update PDS if it is a load process
+  if (PDS_update) {
```

Comment:
This can be simplified (or optimized) further by
`pds(0) += dpstrain(0) - 1/3 * dpvstrain;`
`pds(1) += dpstrain(1) - 1/3 * dpvstrain;`
`pds(2) += dpstrain(2) - 1/3 * dpvstrain;`
`pds(3) += 0.5 * dpstrain(3);`
`pds(4) += 0.5 * dpstrain(4);`
`pds(5) += 0.5 * dpstrain(5);`